gadgets/fsnotify/README.mdx

```diff
@@ -0,0 +1,117 @@
+---
+title: fsnotify
+sidebar_position: 15
+---
+
+# fsnotify
+
+The fsnotify gadget detects applications using inotify or fanotify and enriches the events with the process-related metadata.
+
+## Getting started
+
+Running the gadget:
+
+<Tabs groupId="env">
+    <TabItem value="kubectl-gadget" label="kubectl gadget">
+        ```bash
+        $ kubectl gadget run ghcr.io/inspektor-gadget/gadget/fsnotify:latest [flags]
+        ```
+    </TabItem>
+
+    <TabItem value="ig" label="ig">
+        ```bash
+        $ sudo ig run ghcr.io/inspektor-gadget/gadget/fsnotify:latest [flags]
+        ```
+    </TabItem>
+</Tabs>
+## Flags
+
+### `--fanotify-only`
+
+Show only fanotify events
+
+Default value: ""
+
+### `--inotify-only`
+
+Show only inotify events
+
+Default value: ""
+
+### `--tracee-pid`
+
+Show only fsnotify events generated by processes with this pid
+
+Default value: ""
+
+### `--tracer-group`
+
+Show only fsnotify events generated by a struct fsnotify_group with this address. You can find this address in Golang with kfilefields.ReadPrivateDataFromFd().
+
+Default value: ""
+
+### `--tracer-pid`
+
+Show only fsnotify events generated for processes with this pid
+
+Default value: ""
+
+## Guide
+
+### Example with inotify
+
+Start an application using inotify:
+```bash
+inotifywatch /tmp/
+```
+
+You can generate events in another terminal with:
+```bash
+touch /tmp/ABCDE
+```
+
+The fsnotify-enricher gadget can observe and enrich the inotify events in the following way:
+```bash
+$ sudo ig run ghcr.io/inspektor-gadget/gadget/fsnotify:latest --inotify-only --fields=tracer_comm,tracee_comm,i_wd,i_mask,i_cookie,name
+TRACER_COMM  TRACEE_COMM I_WD I_MASK    I_COOKIE NAME
+inotifywatch touch       1    134217760 0        ABCDE
+inotifywatch touch       1    134217732 0        ABCDE
+inotifywatch touch       1    134217736 0        ABCDE
```

Comment:
Is is expected for NAME not to include the whole path?

Reply:
It's not implemented yet, see: https://github.com/alban/fsnotify-enricher/issues/1

It is not that easy to implement because at the time the kprobes are called, the kernel data structures only contain the name string and the inode number (in the inotify case) but no `struct file` or `struct inode`.